### DISCUSSION TITLE
Adding delay to redirect a route

### DISCUSSION THREAD

**Discussion by oguzhankrcb** [livewire/livewire#8696] [3. Feature ideas]
Hi all,

I was troubling with redirecting to a specific route with some delay, i needed this because i wanted to show some text message to user and then redirect route automatically after `X` seconds. So i came with an idea; I used `js` method of livewire and developed a trait like this:

```
namespace App\Livewire\Concerns;

use Exception;
use Livewire\Component;

trait RedirectsWithDelay
{
    /**
     * @throws Exception
     */
    public function redirectToRouteWithDelay(string $route, int $delayInMilliseconds): void
    {
        if ( ! is_subclass_of($this, Component::class)) {
            throw new Exception(__FUNCTION__ . ' method only works in a Livewire Component!');
        }

        $this->js(
            "setTimeout(function() {
                    window.location.replace('" . $route . "');
                }, " . $delayInMilliseconds . ");"
        );
    }
}
```

And used it like this:

`$this->redirectToRouteWithDelay(route('web.home-page'), 3000);`

It worked like a charm, we can add a similar thing to livewire's code too.

I want to hear about your thoughts and suggestions.

**Comment by lrljoe** [comment]
The thing that immediately jumps out is lack of interface with navigate, so the whole page will re-render rather than the partials.
